package/abs/rancher-abs

```diff
@@ -0,0 +1,464 @@
+                echo "${get_system_available_device_path_result}"
+            fi
+        fi
+    fi
+}
+
+
+# Detach aliyun disk
+# - # https://help.aliyun.com/document_detail/25516.html?spm=5176.doc25515.6.870.LtJDTe
+ALIYUN_DISK_DETACH() {
+    LOG "\t> [disk detach]" /tmp/rancher_abs.log
+
+    local aliyun_options="--InstanceId ${INSTANCE_ID}"
+    local aliyun_call_result=$(aliyuncli ecs DetachDisk ${aliyun_options} --DiskId ${DISK_ID} --output json 2>&1)
+    if [ $? -ne 0 ]; then
+        LOG "\t>> fail, ${aliyun_call_result}" /tmp/rancher_abs.log
+        echo "${aliyun_call_result}"
+        exit 1
+    else
+        LISTENING_ALIYUN_DISK_STATUS "In_use" "Available"
+
+        aliyun_options="--DiskId ${DISK_ID} --Description rancher_detach"
+        aliyun_call_result=$(aliyuncli ecs ModifyDiskAttribute ${aliyun_options} --output json 2>&1)
+        if [ $? -ne 0 ]; then
+            LOG "\t>> fail, ${aliyun_call_result}" /tmp/rancher_abs.log
+            echo "${aliyun_call_result}"
+            exit 1
+        else 
+            LOG "\t>> success" /tmp/rancher_abs.log
+        fi
+    fi
+}
+
+
+# Just init
+init() {
+    GET_INSTANCE_ID
+
+    LOG "\n[init] on ${TRANSCATION_SEQ}" /tmp/rancher_abs.log
+
+    local region_id=$(curl http://100.100.100.200/latest/meta-data/region-id 2>/dev/null)
+
+    mkdir -p /root/.aliyuncli
+    echo -e "[default]\naliyun_access_key_id=${ACCESS_KEY_ID}\naliyun_access_key_secret=${ACCESS_KEY_SECRET}" > /root/.aliyuncli/credentials
+    echo -e "[default]\noutput=json\nregion=${region_id}" > /root/.aliyuncli/configure
+
+    LOG "> success" /tmp/rancher_abs.log
+    print_success
+}
+
+
+# After container has been created with volumne driver "rancher-aliyun-bs"
+# - # https://help.aliyun.com/document_detail/25513.html?spm=5176.doc25514.6.866.Ou38fP
+create() {
+    # - The OPTS variable will be populated from the input JSON as a map of key/values
+    GET_INSTANCE_ID
+
+    LOG "\n[create] on ${TRANSCATION_SEQ}, with instance ${INSTANCE_ID}" /tmp/rancher_abs.log
+
+    # - Check params
+    local zone_id
+    local disk_name
+    local size
+    local disk_category
+    local description
+    local snapshot_id
+    local client_token
+
+    DISK_ID=${OPTS['diskId']}
+    if [ -n "${DISK_ID}" ]; then
+        print_success
+        exit 0
+    fi
+
+    disk_name=${OPTS['diskName']}
+    if [ -z "${disk_name}" ]; then
+        # print_error "diskName is required"
+        disk_name=$(GEN_DISK_NAME)
+    fi
+
+    zone_id=$(curl http://100.100.100.200/latest/meta-data/zone-id 2>/dev/null)
+    size=${OPTS['size']:-"5"}
+    disk_category=${OPTS['diskCategory']:-"cloud"}
+    description=${OPTS['description']}
+    snapshot_id=${OPTS['snapshotId']}
+
+    # - Generate a client_token
+    client_token=$(GEN_CLIENT_TOKEN)
+
+    # - Append command options
+    local aliyun_options="--ZoneId ${zone_id} --DiskName ${disk_name} --Size ${size} --DiskCategory ${disk_category}"
```

Comment:
Inconsistent style. For `zone_id` you separated the define and assignment. Here you define it locally. You should stick to one style. Define it when using is fine with bash.